pkg/operator/ceph/pool/controller.go

```diff
@@ -127,78 +116,108 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to get CephBlockPool")
+		return reconcile.Result{}, errors.Wrapf(err, "failed to get CephBlockPool")
 	}
 
-	// validate the pool settings
-	if err := ValidatePool(r.context, cephBlockPool); err != nil {
-		updateCephBlockPoolStatus(cephBlockPool.GetName(), cephBlockPool.GetNamespace(), k8sutil.ReconcileFailedStatus, r.context)
-		return reconcile.Result{}, errors.Wrapf(err, "invalid pool CR %q spec", cephBlockPool.Name)
+	// The CR was just created, initializing status fields
+	if cephBlockPool.Status == nil {
+		cephBlockPool.Status = &cephv1.Status{}
+		cephBlockPool.Status.Phase = k8sutil.Created
+		err := opcontroller.UpdateStatus(r.client, cephBlockPool)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to set status")
+		}
 	}
 
-	// Set a finalizer so we can do cleanup before the object goes away
-	if !opcontroller.Contains(cephBlockPool.GetFinalizers(), cephBlockPoolFinalizer) {
-		err := r.addFinalizer(cephBlockPool)
-		if err != nil {
-			return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to add finalizer")
+	// Make sure a CephCluster is present otherwise do nothing
+	_, isReadyToReconcile, cephClusterExists, reconcileResponse := opcontroller.IsReadyToReconcile(r.client, r.context, request.NamespacedName)
+	if !isReadyToReconcile {
+		// This handles the case where the Ceph Cluster is gone and we want to delete that CR
+		// We skip the deleteUser() function since everything is gone already
```

Comment:
Should this refer to the pool instead of user?

Reply:
oops!